edit/usw-integration.js

```diff
@@ -1,49 +1,91 @@
-/* global $ $create $remove */// dom.js
+/* global $ $create $remove messageBoxProxy showSpinner toggleDataset */// dom.js
+/* global API msg */// msg.js
+/* global URLS */// toolbox.js
+/* global baseInit */
 /* global editor */
-
+/* global t */// localization.js
 'use strict';
 
-let uswPort;
+(() => {
+  //#region Main
 
-function connectToPort() {
-  if (!uswPort) {
-    uswPort = chrome.runtime.connect({name: 'link-style-usw'});
-    uswPort.onDisconnect.addListener(err => {
-      throw err;
-    });
-  }
-}
+  const ERROR_TITLE = 'UserStyles.world ' + t('genericError');
+  const PROGRESS = '#usw-progress';
+  let spinnerTimer = 0;
+
+  msg.onExtension(request => {
+    if (request.method === 'uswData' &&
+        request.style.id === editor.style.id) {
+      Object.assign(editor.style, request.style);
+      updateUI();
+    }
+  });
 
+  baseInit.ready.then(() => {
+    updateUI();
+    $('#usw-publish-style').onclick = disableWhileActive(publishStyle);
+    $('#usw-disconnect').onclick = disableWhileActive(disconnect);
+  });
+
+  async function publishStyle() {
+    const {id} = editor.style;
+    if (await API.data.has('usw' + id) &&
+        !await messageBoxProxy.confirm(t('publishRetry'), 'danger', ERROR_TITLE)) {
+      return;
+    }
+    const res = await API.usw.publish(id, editor.getValue());
+    const title = `${new Date().toLocaleString()}\n${res}`;
+    $(PROGRESS).append(.../^Error:/.test(res) ? [
+      $create('div.error', {title}, res),
+      $create('div', t('publishReconnect')),
+    ] : [
+      $create('span.success', {title}, '\u2713'), // checkmark
+    ]);
+  }
 
-/* exported revokeLinking */
-function revokeLinking() {
-  connectToPort();
+  async function disconnect() {
+    await API.usw.revoke(editor.style.id);
+  }
 
-  uswPort.postMessage({reason: 'revoke', data: editor.style});
-}
+  function updateUI(style = editor.style) {
+    const usw = style._usw || {};
+    const section = $('#publish');
+    toggleDataset(section, 'connected', usw.token);
+    for (const type of ['name', 'description']) {
+      const el = $(`dd[data-usw="${type}"]`, section);
+      el.textContent = el.title = usw[type] || '';
+    }
+    const elUrl = $('#usw-url');
+    elUrl.href = `${URLS.usw}${usw.id ? `style/${usw.id}` : ''}`;
+    elUrl.textContent = elUrl.textContent.replace(/#\d+/, '').replace(elUrl.hostname, '$&#' + usw.id);
```

Comment:
```suggestion
    elUrl.textContent = elUrl.textContent.replace(/#\d+/, '').replace(elUrl.hostname, '$&' + (usw.id ? `#${usw.id}` : ''));
```
Otherwise you will see `Using <userstyles.world#undefined>`